### PR TITLE
[core]Correct Constant creation from string

### PR DESCRIPTION
**Pull request by praasz**
### Details:
 - Create Constant from string using different type for floating-point, integral signed and integral unsigned to avoid precision loss.
 - Minor optimization in values and shape size check to reduce binary size.

### Tickets:
 - [CVS-124982](https://jira.devtools.intel.com/browse/CVS-124982)

### CODE REVIEW

File: src/core/src/op/constant.cpp [hunk 1]
```diff
@@ -43,12 +43,23 @@ std::string to_cpp_string(T value) {
     }
 }
 
-std::vector<double> from_string_vector(const std::vector<std::string>& str_values) {
-    std::vector<double> values;
+template <class T, typename std::enable_if<std::is_integral<T>::value>::type* = nullptr>
+T str_to_value(const std::string& s, size_t* pos) {
+    return static_cast<T>(std::is_signed<T>::value ? std::stol(s, pos) : std::stoul(s, pos));
+}
```
Comment (t-jankowski) [inline src/core/src/op/constant.cpp:49]: What about having it separate?
```suggestion
template <class T,
          typename std::enable_if<std::is_integral<T>::value>::type* = nullptr,
          typename std::enable_if<std::is_signed<T>::value>::type* = nullptr>
T str_to_value(const std::string& s, size_t* pos) {
    return static_cast<T>(std::stol(s, pos));
}

template <class T,
          typename std::enable_if<std::is_integral<T>::value>::type* = nullptr,
          typename std::enable_if<std::is_unsigned<T>::value>::type* = nullptr>
T str_to_value(const std::string& s, size_t* pos) {
    return static_cast<T>(std::stoul(s, pos));
}
```

Comment (praasz) [reply src/core/src/op/constant.cpp:49]: As for internal function there is not much benefit re-work it, Will keep current solution (the inside if is well optimized as value is resolved during compile time).

**Review by t-jankowski — APPROVED** [review]

**Review by jane-intel — APPROVED** [review]

**Review by slyalin — COMMENTED** [review]
To save time on resolving of hard conflicts, please make sure that code is aligned with https://github.com/openvinotoolkit/openvino/pull/21244 in general.

**Comment by rkazants** [comment]
@praasz, let us postpone merge of this PR

**Comment by praasz** [comment]
> @praasz, please let us postpone merge of this PR. Let us merge string tensor support first. I will finish this today. Hope this PR is not critical, really sorry for this.

I can wait and do sync (resolve conflicts) with master after #21244 merge

**Comment by praasz** [comment]
@slyalin , @rkazants PR synced. Could you check.

**Review by rkazants — APPROVED** [review]

**Comment by rkazants** [comment]
May be it makes sense to add unit-test to demonstrate that now numeric accuracy is not lost for integer types for example.

**Comment by praasz** [comment]
> May be it makes sense to add unit-test to demonstrate that now numeric accuracy is not lost for integer types for example.

Simple tests added.

**Pull request merged**